client/msw-handlers.ts

```diff
@@ -1112,6 +1112,9 @@ const handler =
       if (typeof thrown === "string") {
         return json({ message: thrown }, { status: 400 });
       }
+      if (thrown instanceof Response) {
+        return thrown;
+      }
```

Comment:
can we allow this in the regular return value too? does that make the return types on the handler functions horrible?